opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -1205,6 +1223,13 @@ void EclipseWriter::writeInit(const SimulatorTimerInterface &timer)
             EclipseWriterDetails::restrictAndReorderToActiveCells(data, gridToEclipseIdx_.size(), gridToEclipseIdx_.data());
             fortio.writeKeyword("PERMZ", data);
         }
+        if (nnc.hasNNC()) {
+            std::vector<double> tran;
```

Comment:
> As you can see three lines up, the other data written are also double.

You are right - the `writeKeyword( )` function actually does the `double -> float` conversion.